src/main/java/Todo.java

```diff
@@ -1,11 +1,20 @@
 public class Todo extends Task {
 
-  public Todo(String desc) {
-    super(desc);
-  }
+    public Todo(String desc) {
+        super(desc);
+    }
 
-  @Override
-  public String toString() {
-      return "[T]" + getStatusIcon() + " " + desc;
-  }
+    static public Todo parse(String input) throws DukeException {
+        String body = input.split("todo")[1].strip();
+        if (!body.trim().isEmpty()) {
+            return new Todo(body);
+        } else {
+            throw new DukeException("Todo description cannot be empty");
+        }
+    }
+
+    @Override
+    public String toString() {
+        return "[T]" + getStatusIcon() + " " + desc;
```

Comment:
Can store the Task type as an attribute so you dun hard code, e.g., taskType = "T"